libbpf-rs/src/program.rs

```diff
@@ -1034,6 +1053,120 @@ impl<'obj> ProgramMut<'obj> {
         Ok(link)
     }
 
+    /// Attach this program to multiple
+    /// [uprobes](https://www.kernel.org/doc/html/latest/trace/uprobetracer.html) at once.
+    pub fn attach_uprobe_multi(
+        &self,
+        pid: i32,
+        binary_path: impl AsRef<Path>,
+        func_pattern: impl AsRef<str>,
+        retprobe: bool,
+        session: bool,
+    ) -> Result<Link> {
+        let opts = UprobeMultiOpts {
+            syms: Vec::new(),
+            offsets: Vec::new(),
+            ref_ctr_offsets: Vec::new(),
+            cookies: Vec::new(),
+            retprobe,
+            session,
+            _non_exhaustive: (),
+        };
+
+        self.attach_uprobe_multi_with_opts(pid, binary_path, func_pattern, opts)
+    }
+
+    /// Attach this program to multiple
+    /// [uprobes](https://www.kernel.org/doc/html/latest/trace/uprobetracer.html)
+    /// at once, providing additional options.
+    pub fn attach_uprobe_multi_with_opts(
+        &self,
+        pid: i32,
+        binary_path: impl AsRef<Path>,
+        func_pattern: impl AsRef<str>,
+        opts: UprobeMultiOpts,
+    ) -> Result<Link> {
+        let path = util::path_to_cstring(binary_path)?;
+        let path_ptr = path.as_ptr();
+
+        let UprobeMultiOpts {
+            syms,
+            offsets,
+            ref_ctr_offsets,
+            cookies,
+            retprobe,
+            session,
+            _non_exhaustive,
+        } = opts;
+
+        let pattern = util::str_to_cstring(func_pattern.as_ref())?;
+        let pattern_ptr = pattern.as_ptr();
+
+        let syms_cstrings = if !syms.is_empty() {
+            Some(
```

Comment:
Why do we need an `Option` here? It seems we should be able to use the same pattern we use for all other vectors, no?